diff -up gnupg-2.0.14/g10/gpgv.c.libgcrypt-init gnupg-2.0.14/g10/gpgv.c
--- gnupg-2.0.14/g10/gpgv.c.libgcrypt-init	2009-09-21 18:53:51.000000000 +0200
+++ gnupg-2.0.14/g10/gpgv.c	2014-05-07 16:13:57.546261285 +0200
@@ -149,7 +149,14 @@ main( int argc, char **argv )
   /* Make sure that our subsystems are ready.  */
   i18n_init();
   init_common_subsystems ();
-  
+
+  if (!gcry_check_version (NEED_LIBGCRYPT_VERSION) )
+    {
+      log_fatal ( _("%s is too old (need %s, have %s)\n"), "libgcrypt",
+                  NEED_LIBGCRYPT_VERSION, gcry_check_version (NULL) );
+    }
+  gcry_control (GCRYCTL_DISABLE_SECMEM, 0);
+
   gnupg_init_signals (0, NULL);
 
   opt.command_fd = -1; /* no command fd */
@@ -163,7 +170,9 @@ main( int argc, char **argv )
   tty_no_terminal(1);
   tty_batchmode(1);
   disable_dotlock();
-  
+
+  gcry_control (GCRYCTL_INITIALIZATION_FINISHED, 0);
+
   pargs.argc = &argc;
   pargs.argv = &argv;
   pargs.flags=  1;  /* do not remove the args */
